plutus-chain-index/src/Plutus/ChainIndex/Events.hs

```diff
@@ -33,13 +33,21 @@ period = 2_000_000 -- 2s
 -- By doing this we accumulate some number of blocks but with less than 'queueSize' number of transactions.
 -- This approach helps to process blocks with a constant memory usage.
 --
+-- However, once we are in sync with the node, we want to process every block
+-- instead of batches of blocks so that we can update the database as quickly
+-- as possible.
+--
 -- Just accumulating 'queueSize' blocks doesn't work as a block can have any number of transactions.
 -- It works fine at the beginning of the chain but later blocks grow in their size and the memory
 -- usage grows tremendously.
-measureEventByTxs :: ChainSyncEvent -> Natural
-measureEventByTxs = \case
-  (RollForward (CI.Block _ transactions) _) -> fromIntegral $ length transactions
-  _                                         -> 1
+measureEventQueueSizeByTxs :: Natural -> ChainSyncEvent -> Natural
+measureEventQueueSizeByTxs maxQueueSize (RollForward (CI.Block syncTip transactions) nodeTip) =
+    let syncState = getSyncState (tipAsPoint syncTip) (tipAsPoint nodeTip)
+        txLen = fromIntegral $ length transactions
+     in if isSyncStateSynced syncState
+           then max (maxQueueSize + 1) txLen
```

Comment:
I think that `max` adds ambiguity here.

If we are synced, we want to exit asap just like with the resume and rollback case. So we shouldn't care at all about `txLen`. If `txLen` is bigger than `maxQueueSize + 1`, it will behave in the same way. But this `max` makes to think more about the code and question what logic it handles.

Reply:
Ah ok! Then just `maxQueueSize + 1`?

Reply:
Ah ok! You mean the following change?

```suggestion
           then maxQueueSize + 1
```

Reply:
Yeah, that's right 👍